extensions/ql-vscode/src/common/file-tree-nodes.ts

```diff
@@ -91,12 +99,12 @@ export class FileTreeDirectory extends FileTreeNode {
 /**
  * A single file.
  */
-export class FileTreeLeaf extends FileTreeNode {
-  constructor(_path: string, _name: string) {
-    super(_path, _name);
+export class FileTreeLeaf<T = undefined> extends FileTreeNode<T> {
+  constructor(_path: string, _name: string, _data: T) {
```

Comment:
```suggestion
  constructor(_path: string, _name: string, _data?: T) {
```
If we make the `_data` arg optional here then we can revert the change to `qltest-discovery.ts` and avoid changing that file at all in this PR, but not change behaviour.